nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 Angsa Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.hpp"
+#include "nav2_behavior_tree/bt_utils.hpp"
+#include "tf2/utils.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+
+namespace nav2_behavior_tree
+{
+
+RemoveInCollisionGoals::RemoveInCollisionGoals(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::ActionNodeBase(name, conf),
+  initialized_(false),
+  which_costmap_("both"),
+  cost_threshold_(253)
+{}
+
+void RemoveInCollisionGoals::initialize()
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+  get_global_cost_client_ = node_->create_client<nav2_msgs::srv::GetCost>(
+    "global_costmap/get_cost_global_costmap");
+  get_local_cost_client_ = node_->create_client<nav2_msgs::srv::GetCost>(
```

Comment:
We have a simple service client in Nav2 utils that you may want to use instead, it reduces alot of the boilerplate duplication and largely use it across nav2's codebase

Reply:
> and largely use it across nav2's codebase

I guess you mean [this](https://github.com/ros-navigation/navigation2/blob/main/nav2_util/include/nav2_util/service_client.hpp). It makes me realize that something in my implementation (and using the service_client) is not ideal -> if the service takes too long it would block the BT ticking. 

This is why [this logic](https://github.com/ros-navigation/navigation2/blob/9e90417dd57b27c99c29b849321ad90205ca5523/nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp#L205) exists in "pure" BT service nodes.

Do you think for BT nodes that call a service we should reimplement similar logic that would return RUNNING if the service call takes too long? I would think so